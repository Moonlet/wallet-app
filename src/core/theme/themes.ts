export enum Theme {
    dark = 'dark',
    light = 'light'
}
