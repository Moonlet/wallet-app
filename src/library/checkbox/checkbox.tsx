import React from 'react';
import { TouchableOpacity } from 'react-native';
import { Text } from '../index';

import stylesProvider from './styles';
import { withTheme } from '../../core/theme/with-theme';
import { Icon } from '../../components/icon';

export interface ICheckboxProps {
    children?: any;
    style?: any;
    text?: string;
    disabled?: boolean;
    checked?: boolean;
    onPress?: any;
    styles: ReturnType<typeof stylesProvider>;
}

export const CheckboxComponent = (props: ICheckboxProps) => (
    <TouchableOpacity
        disabled={props.disabled}
        onPress={props.onPress}
        style={[props.styles.container, props.style]}
    >
        <Icon
            name={props.checked ? 'check-2-thicked' : 'check-2'}
            size={18}
            style={props.styles.icon}
        />
        {props.text && <Text style={props.styles.text}>{props.text}</Text>}
    </TouchableOpacity>
);

export const Checkbox = withTheme(stylesProvider)(CheckboxComponent);
