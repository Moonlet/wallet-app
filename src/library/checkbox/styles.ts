import { StyleSheet } from 'react-native';
import { ITheme } from '../../core/theme/itheme';
import { BASE_DIMENSION, normalize } from '../../styles/dimensions';

export default (theme: ITheme) =>
    StyleSheet.create({
        container: {
            flex: 1,
            flexDirection: 'row'
        },
        icon: {
            color: theme.colors.accent,
            paddingRight: BASE_DIMENSION,
            paddingTop: BASE_DIMENSION / 4
        },
        text: {
            fontSize: normalize(17),
            lineHeight: normalize(22),
            color: theme.colors.textSecondary
        }
    });
