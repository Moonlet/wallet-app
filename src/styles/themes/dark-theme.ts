import { ITheme } from '../../core/theme/itheme';
import { COLORS } from '../colors';

export const darkTheme: ITheme = {
    dark: true,
    shadowGradient: ['#00000000', '#000000CC'],

    fontSize: {
        small: 12,
        regular: 16,
        large: 34
    },

    colors: {
        primary: COLORS.GRAY,
        accent: COLORS.AQUA,
        text: COLORS.WHITE,
        textDarker: COLORS.LIGHT_GRAY,
        positive: COLORS.HOT_GREEN,
        negative: COLORS.HOT_RED,

        cardBackground: COLORS.DARK_GRAY,
        appBackground: COLORS.COD_GRAY,
        settingsDivider: COLORS.BLACK
    }
};
