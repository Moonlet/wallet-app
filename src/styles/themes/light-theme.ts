import { ITheme } from '../../core/theme/itheme';
import { darkTheme } from './dark-theme';

export const lightTheme: ITheme = darkTheme;
