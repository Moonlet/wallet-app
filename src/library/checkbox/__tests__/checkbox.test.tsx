import React from 'react';
import { CheckboxComponent, ICheckboxProps } from '../checkbox';
import { darkTheme } from '../../../styles/themes/dark-theme';
import styleProvider from '../styles';

import { shallow } from 'enzyme';

const props: ICheckboxProps = {
    styles: styleProvider(darkTheme),
    text: 'agree'
};

describe('library checkbox', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<CheckboxComponent {...props} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
