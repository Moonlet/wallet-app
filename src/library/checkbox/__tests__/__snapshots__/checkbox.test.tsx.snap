// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`library checkbox renders correctly 1`] = `
"<TouchableOpacity disabled={[undefined]} onPress={[undefined]} style={{...}} activeOpacity={0.2}>
  <Icon name=\\"check-2\\" size={33} style={{...}} allowFontScaling={false} />
  <withTheme(TextComponent) style={{...}}>
    agree
  </withTheme(TextComponent)>
</TouchableOpacity>"
`;
